pkg/util/kube.go

```diff
@@ -419,6 +419,38 @@ func GetPodName(n, labelSelector string) (pod string, err error) {
 	return
 }
 
+// Check PodReady (Example: 2/2 container running on a pod) returns true if the pod is ready. Params: namespace, pod label and timeout to check ready pod.
+func CheckPodReady(n, labelSelector string, timeout int) (ready bool, err error) {
+	ready = false
+	for i := 0; i < timeout; i++ {
+		pod, err := GetPodName(n, labelSelector)
+		status, _ := Shell(`oc get pod %s -n istio-system | grep istiod | awk '{print $2}'`, pod)
+		if err != nil {
+			return ready, err
+		}
+		//split status to get the number of containers expected to be running
+		s := strings.Split(status, "/")
+		//trim the status to get the number of containers running without spaces
+		s[0] = strings.Trim(s[0], "\r\n")
+		s[1] = strings.Trim(s[1], "\r\n")
```

Comment:
This won't be needed if you will use the command I suggested.

Reply:
Deleted